src/SSHDebugPS/VSOperationWaiter.cs

```diff
@@ -71,18 +45,12 @@ public static bool Wait(string actionName, bool throwOnCancel, Action action)
                         {
                             throw new OperationCanceledException();
                         }
-
-                        return false;
-                    }
-
-                    if (t.IsFaulted)
-                    {
-                        throw t.Exception.InnerException;
                     }
+                    success = true;
                 }
-            }
+            );
 
-            return true;
+            return success;
```

Comment:
As you have it, I believe `success` is guaranteed to be `true`, so I'm not sure why you bother having the local variable at all (or return `bool` from this method). It used to be that you'd return `false`  when OCE was thrown and `throwOnCancel==false`, but now you set `success=true` before returning it.

Reply:
I was hoping that if throwOnCancel was true and an exception was thrown or cancel was set, the async method would throw and never set success to true.

Reply:
It would -- but you have code here to catch that exception and swallow it if `!throwOnCancel`.

Reply:
Oh, also: your `action` doesn't actually even accept a `CancellationToken`, so in fact the way you're coding this up, you're offering the user a Cancel button which is a no-op. It would only cancel if the user happened to click it before the `action` delegate were even invoked, which is *highly* unlikely. You should either call the `JTF.Run` overload that does not provide the delegate with a token, or modify your `Wait` method to take an `Action<CancellationToken>` so that the delegates can respond to cancellation requests.